githttp.go

```diff
@@ -10,6 +10,14 @@ import (
 	"strings"
 )
 
+type ResolveInfo struct {
+	Path    string
+	Request *http.Request
+}
```

Comment:
I was following the pattern that the auth info uses of having a struct, which allows for future extension without breaking the API. I was thinking that ideally it would be nice to have a context dictionary in the struct instead of passing request all the way through the layers, but wasn't obvious how to do that without a larger refactor since the auth is it's own independent middleware.